osu.Server.Spectator.Tests/Multiplayer/MultiplayerAllPlayersRoundRobinQueueTests.cs

```diff
@@ -91,6 +66,200 @@ public async Task RoundRobinOrdering()
             await checkCurrentItem(4);
         }
 
+        [Fact]
+        public async Task RoundRobinOrderingWithManyUsers()
+        {
+            Database.Setup(d => d.GetBeatmapChecksumAsync(3333)).ReturnsAsync("3333");
+
+            CreateUser(1, out var contextUser1, out _);
+            CreateUser(2, out var contextUser2, out _);
+            CreateUser(3, out var contextUser3, out _);
+            CreateUser(4, out var contextUser4, out _);
+
+            // ---
+            // User 1 joins and begins adding items.
+            // ---
+
+            SetUserContext(contextUser1);
+            await Hub.JoinRoom(ROOM_ID);
+            await Hub.ChangeSettings(new MultiplayerRoomSettings { QueueMode = QueueMode.AllPlayersRoundRobin });
+
+            // | item | user |
+            // | ---- | ---- |
+            // |  1   |  1   | (new)
+            await checkOrder(1);
+
+            // | item | user |
+            // | ---- | ---- |
+            // |  1   |  1   |
+            // |  2   |  1   | (new)
+            await addItem();
+            await checkOrder(1, 2);
+
+            // | item | user |
+            // | ---- | ---- |
+            // |  1   |  1   |
+            // |  2   |  1   |
+            // |  3   |  1   | (new)
+            await addItem();
+            await checkOrder(1, 2, 3);
+
+            // ---
+            // User 2 joins and begins adding items.
+            // ---
+
+            SetUserContext(contextUser2);
+            await Hub.JoinRoom(ROOM_ID);
+
+            // | item | user |
+            // | ---- | ---- |
+            // |  1   |  1   |
+            // |  4   |  2   | (new)
+            // |  2   |  1   |
+            // |  3   |  1   |
+            await addItem();
+            await checkOrder(1, 4, 2, 3);
+
+            // ---
+            // User 3 joins and begins adding items.
+            // ---
+
+            SetUserContext(contextUser3);
+            await Hub.JoinRoom(ROOM_ID);
+
+            // | item | user |
+            // | ---- | ---- |
+            // |  1   |  1   |
+            // |  4   |  2   |
+            // |  5   |  3   | (new)
+            // |  2   |  1   |
+            // |  3   |  1   |
+            await addItem();
+            await checkOrder(1, 4, 5, 2, 3);
```

Comment:
Of importance, this is the first test that fails on master, showing how https://github.com/ppy/osu-server-spectator/issues/126 was fixed in this PR.

The order on master is `1 5 4 2 3`, so item `5` by the new user was inserted towards the top of the queue, in front of an existing one of the same "priority" by another user.